lib/ace/mode/latex_parser.js

```diff
@@ -1,23 +1,42 @@
 define(function(require, exports, module) {
 "use strict";
 
-var WORD_PATTERN = "[a-zA-Zа-яА-ЯёЁ]+";
-var WORD_REGEX = new RegExp("^" + WORD_PATTERN);
+var WORD_REGEX = /^[a-zа-яё]+/i,
+    // % * <foo@bar.com> 2015-01-12T00:17:27.829Z:
+    COMMENT_REGEXP = /^\s*%\s([\*\^])\s\<(.+)\>\s(.+):$/;
 
 var LatexParser = exports.LatexParser = function(bgTokenizer) {
     this.myErrors = [];
+    this.myComments = [];
     this.myTokenizer = bgTokenizer;
 };
 
 (function(){
+    this.setSpellingCheckDictionary = function(dictionary) {
+        this.myDictionary = dictionary;
+    };
 
-    this.go = function(doc, dictionary){
-        if (dictionary == null) {
+    /**
+     * Parse the document
+     * @param doc           document object
+     * @param doSpellcheck  true if we want spellcheck
+     * @param doGetComments true if we want get text comments
+     */
+    this.go = function(doc, doSpellcheck, doGetComments) {
+        if (doSpellcheck && this.myDictionary == null) {
+            // can't spellcheck without dictionary
+            doSpellcheck = false;
+        }
+        if (!doSpellcheck && !doGetComments) {
+            // there is nothing to do
             return;
         }
+
         var lines = doc.getAllLines();
 
         this.myErrors = [];
+        this.myComments = [];
```

Comment:
maybe "user comments" or "discussions"? Just to distinguish them from generic latex comments?